packages/gafl-webapp-service/src/pages/licence-details/licence-start-time/__tests__/route.spec.js

```diff
@@ -137,3 +144,79 @@ describe('licenceStartTimeRoute > onPostAuth', () => {
     })
   })
 })
+
+describe('getData', () => {
+  const getData = pageRoute.mock.calls[1][4]
+  const mockTransactionCacheGet = jest.fn()
+  const mockRequest = {
+    cache: () => ({
+      helpers: {
+        transaction: {
+          getCurrentPermission: mockTransactionCacheGet
+        }
+      }
+    }),
+    locale: 'en-gb'
+  }
+
+  it('startDateStr is result of moment().format()', async () => {
+    const formatSymbol = Symbol('formatResult')
+    moment.mockImplementation(
+      getMomentMockImpl({
+        format: () => formatSymbol
+      })
+    )
+    mockTransactionCacheGet.mockImplementationOnce(() => ({ licenceStartDate: '2020-01-01' }))
+    const { startDateStr } = await getData({
+      cache: () => ({
+        helpers: {
+          transaction: {
+            getCurrentPermission: mockTransactionCacheGet
+          }
+        }
+      }),
+      locale: 'anywhere'
+    })
+    expect(startDateStr).toEqual(formatSymbol)
+  })
+
+  it.each([['2020-07-06'], ['2020-01-01'], ['2022-07-09']])('permission licence start date is passed to moment', async licenceStartDate => {
+    mockTransactionCacheGet.mockImplementationOnce(() => ({ licenceStartDate }))
+    moment.mockImplementation(getMomentMockImpl())
+    await getData(mockRequest)
+    expect(moment).toHaveBeenCalledWith(licenceStartDate, 'YYYY-MM-DD', 'en-gb')
+  })
+
+  it('cacheDateFormat being what is expected', async () => {
+    mockTransactionCacheGet.mockImplementationOnce(() => ({ licenceStartDate: '2020-01-01' }))
+    moment.mockImplementation(getMomentMockImpl())
+    await getData(mockRequest)
+    expect(moment).toHaveBeenCalledWith('2020-01-01', 'YYYY-MM-DD', 'en-gb')
+  })
+
+  it('dddd, Do MMMM, YYYY is passed to format', async () => {
+    mockTransactionCacheGet.mockImplementationOnce(() => ({ licenceStartDate: '2020-01-01' }))
+    const format = jest.fn()
+    moment.mockImplementation(getMomentMockImpl({ format }))
+    await getData(mockRequest)
+    expect(format).toHaveBeenCalledWith('dddd, Do MMMM, YYYY')
+  })
+
+  it.each([['en-gb'], ['cy']])('locale is set on moment, to whatever the request.locale is', async language => {
+    const mockRequest = {
+      cache: () => ({
+        helpers: {
+          transaction: {
+            getCurrentPermission: mockTransactionCacheGet
+          }
+        }
+      }),
+      locale: language
+    }
+    mockTransactionCacheGet.mockImplementationOnce(() => ({ licenceStartDate: '2020-01-01' }))
+    const format = jest.fn()
+    moment.mockImplementation(getMomentMockImpl({ format }))
+    await getData(mockRequest)
+    expect(moment).toHaveBeenCalledWith('2020-01-01', 'YYYY-MM-DD', language)
```

Comment:
Ah, I see you're testing the locale here - good stuff, but as I say you don't care what the first and second arguments are, so use `expect.any(String)`, then use a `Symbol` for locale so you won't need several test cases.

We will need one final test, to check the return value of moment().format() is passed back as `startDateStr`

Reply:
pushed all changes, tests for return value of moment().format() is passed back as startDateStr already exists line 168